controllers/propagator/policy_controller.go

```diff
@@ -160,6 +174,20 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return reconcile.Result{RequeueAfter: duration}, nil
 		}
 
+		// Collect the hub template watch metric; If there's an error, just log it
+		hubTempWatches := r.DynamicWatcher.GetWatchCount()
+		log.V(3).Info("Setting hub template watch metric", "value", hubTempWatches)
+
+		gaugeMetric, err := hubTemplateActiveWatchesMetric.GetMetricWith(prometheus.Labels{
```

Comment:
`GetWatchCount` is for *all* watches and not just the watches for that particular policy.

I do like your idea of being able to distinguish which policy has a lot of watches. Perhaps we could add a method to the dynamic watcher library to accept an `ObjectIdentifier` to get the watch count?

Reply:
The latest changes I have locally use Gauge instead of GaugeVec. I think I need to focus time on my other issues at this point, especially since the cleanup of GaugeVec becomes a little sticky. 🙂